Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -390,594 +174,219 @@ CGImageRef CGImageMaskCreate(size_t width,
-                                                      bitsPerComponent,
-                                                      bitsPerPixel >> 3,
-                                                      width,
-                                                      height,
-                                                      bytesPerRow,
-                                                      data,
-                                                      format);
-
-        newImage = new CGBitmapImage(surfaceInfo);
-    } else {
-        __CGSurfaceInfo surfaceInfo = _CGSurfaceInfoInit(width, height, _ColorBGR);
-
-        newImage = new CGBitmapImage(surfaceInfo);
-        void* pData = newImage->Backing()->LockImageData();
-        int stride = newImage->Backing()->BytesPerRow();
-
-        char* curPosOut = (char*)pData;
-        char* curPosIn = (char*)data;
-        for (unsigned y = 0; y < height; y++) {
-            char* curLineOut = curPosOut;
-            char* curLineIn = curPosIn;
-            for (unsigned x = 0; x < width; x++) {
-                BYTE c = *curLineIn;
-
-                if (c <= ((__CGColorSpace*)colorSpace)->lastColor) {
-                    BYTE* palette = (BYTE*)&((__CGColorSpace*)colorSpace)->palette[c * 3];
-                    *(curLineOut++) = *(palette++);
-                    *(curLineOut++) = *(palette++);
-                    *(curLineOut++) = *(palette++);
-                } else {
-                    *(curLineOut++) = 0;
-                    *(curLineOut++) = 0;
-                    *(curLineOut++) = 0;
-                }
-                curLineIn++;
-            }
-
-            curPosOut += stride;
-            curPosIn += width;
-        }
-
-        newImage->Backing()->ReleaseImageData();
-    }
+CGImageRef CGImageCreateWithPNGDataProvider(CGDataProviderRef source,
+                                            const CGFloat decode[],
+                                            bool shouldInterpolate,
+                                            CGColorRenderingIntent intent) {
+    RETURN_NULL_IF((source == nullptr) || ![(NSObject*)source isKindOfClass:[NSData class]]);
 
-    newImage->_provider = dataProvider;
+    NSData* sourceData = (__bridge NSData*)source;
+    CGImageRef imageRef = _CGImageLoadPNG((void*)[sourceData bytes], [sourceData length]);
 
-    if (colorSpaceAllocated == true) {
-        CGColorSpaceRelease(colorSpace);
-    }
+    RETURN_NULL_IF(!imageRef);
+    imageRef->SetInterpolate(shouldInterpolate).SetRenderingIntent(intent);
 
-    return (CGImageRef)newImage;
+    return imageRef;
 }
 
-static void PNGWriteFunc(png_structp png_ptr, png_bytep data, png_size_t length) {
-    id dataOut = (id)png_get_io_ptr(png_ptr);
-
-    [dataOut appendBytes:data length:length];
+/**
+ @Status Stub
+ @Notes
+*/
+CGImageRef CGImageCreateCopyWithColorSpace(CGImageRef ref, CGColorSpaceRef colorSpace) {
+    RETURN_NULL_IF(!ref);
+    UNIMPLEMENTED();
```

Comment:
the unimplemented is for the analyzer, this is prep code.
Also a caller would not be able to make this call given we will have a stubbed method.

---
In reply to: [84549825](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84549825) [](ancestors = 84549825)